packages/video_player/video_player_avfoundation/darwin/Classes/FVPVideoPlayerPlugin.m

```diff
@@ -558,12 +551,29 @@ - (void)setPlaybackSpeed:(double)speed {
 }
 
 - (CVPixelBufferRef)copyPixelBuffer {
+  CVPixelBufferRef buffer = NULL;
   CMTime outputItemTime = [_videoOutput itemTimeForHostTime:CACurrentMediaTime()];
   if ([_videoOutput hasNewPixelBufferForItemTime:outputItemTime]) {
-    return [_videoOutput copyPixelBufferForItemTime:outputItemTime itemTimeForDisplay:NULL];
+    buffer = [_videoOutput copyPixelBufferForItemTime:outputItemTime itemTimeForDisplay:NULL];
   } else {
-    return NULL;
+    // If the current time isn't available yet, use the time that was checked when informing the
+    // engine that a frame was available (if any).
+    CMTime lastAvailableTime = self.frameUpdater.lastKnownAvailableTime;
+    if (CMTIME_IS_VALID(lastAvailableTime)) {
+      buffer = [_videoOutput copyPixelBufferForItemTime:lastAvailableTime itemTimeForDisplay:NULL];
+    }
+  }
+
+  if (self.waitingForFrame && buffer) {
```

Comment:
When will buffer be nil (seeking to the end?)
 
should this be just `if (self.waitingForFrame)` so that displayLink is guaranteed to shutdown?

Reply:
> When will buffer be nil (seeking to the end?)

Seeking to the end should work. It could be nil on iOS since iOS just fires the display link all the time and relies on the engine (apparently) allowing null frames. There may also be edge cases on macOS where we still can get nulls.

> should this be just `if (self.waitingForFrame)` so that displayLink is guaranteed to shutdown?

But if we're waiting for a frame and we didn't get one yet, then we need to wait longer.

Consider iOS seeking while paused, for instance, and then firing the display link indiscriminately before the video frame is actually ready. In that case, we don't want to stop the display link or the frame will be stuck on the old location. Instead we need to wait until the video decoding catches up and there's a frame to show.

Reply:
It may be worth documenting the macOS edge cases here in a comment.

Reply:
To clarify, I don't actually know what the macOS edge cases are. Recent comments in https://github.com/flutter/flutter/issues/135999 suggest they still exist, but I don't yet know when/why.